Mathlib/NumberTheory/Padics/PadicVal.lean

```diff
@@ -535,6 +585,23 @@ theorem padicValNat_primes {q : ℕ} [hp : Fact p.Prime] [hq : Fact q.Prime] (ne
     (not_congr (Iff.symm (prime_dvd_prime_iff_eq hp.1 hq.1))).mp neq
 #align padic_val_nat_primes padicValNat_primes
 
+/-- The p-adic valuation of `n` is less than or equal to its logarithm w.r.t `p`.-/
+lemma le_nat_log (n : ℕ) [hp : Fact (Nat.Prime p)] : padicValNat p n ≤ Nat.log p n := by
+  rcases eq_or_ne n 0 with rfl | h
+  · rw [padicValNat.zero, Nat.log_zero_right]
+  · refine' Nat.le_log_of_pow_le hp.elim.one_lt _
+    exact le_of_dvd (Nat.pos_of_ne_zero h) pow_padicValNat_dvd
+
+lemma le_nat_log_of_le {n₁ n₂ : ℕ} [Fact (Nat.Prime p)] (hn : n₁ ≤ n₂) :
```

Comment:
These last two lemmas are no longer used.

Reply:
Yeah but they'd be good to have around.

Reply:
The second one seems potentially useful, but this first lemma just gives a weaker upper bound than the preceding lemma, I think?

Reply:
I use the first lemma in the proof of the second, so might as well factor it out into its own lemma?

Reply:
Sorry, I meant that `le_nat_log_of_le` doesn't improve on the upper bound in `padicValNat_le_nat_log`, so I'm not sure whether it makes sense to keep `le_nat_log_of_le`.

Reply:
Ah okay, I agree. I just removed it.